models/classes/runner/time/QtiTimeLine.php

```diff
@@ -0,0 +1,306 @@
+     * @since 5.1.0
+     */
+    public function serialize()
+    {
+        $data = [];
+
+        foreach ($this->points as $point) {
+            $data [] = serialize($point);
+        }
+
+        return serialize($data);
+    }
+
+    /**
+     * Constructs the object
+     * @link http://php.net/manual/en/serializable.unserialize.php
+     * @param string $serialized <p>
+     * The string representation of the object.
+     * </p>
+     * @return void
+     * @since 5.1.0
+     */
+    public function unserialize($serialized)
+    {
+        $data = unserialize($serialized);
+
+        if ($data && is_array($data)) {
+            foreach ($data as $serializedPoint) {
+                $this->add(unserialize($serializedPoint));
+            }
+        }
+    }
+
+    /**
+     * Gets the list of TimePoint present in the TimeLine
+     * @return array
+     */
+    public function getPoints()
+    {
+        return $this->points;
+    }
+
+
+    /**
+     * Adds another TimePoint inside the TimeLine
+     * @param TimePoint $point
+     * @return TimeLine
+     */
+    public function add(TimePoint $point)
+    {
+        $this->points[] = $point;
+        return $this;
+    }
+
+    /**
+     * Removes all TimePoint corresponding to the provided criteria
+     * @param string|array $tag A tag or a list of tags to filter
+     * @param int $target The type of target TimePoint to filter
+     * @param int $type The tyoe of TimePoint to filter
+     * @return int Returns the number of removed TimePoints
+     */
+    public function remove($tag, $target = TimePoint::TARGET_ALL, $type = TimePoint::TYPE_ALL)
+    {
+        $tags = is_array($tag) ? $tag : [$tag];
+        $removed = 0;
+        foreach ($this->points as $idx => $point) {
+            if ($point->match($tags, $target, $type)) {
+                unset($this->points[$idx]);
+                $removed++;
+            }
+        }
+        return $removed;
+    }
+
+    /**
+     * Clears the TimeLine from all its TimePoint
+     * @return TimeLine
+     */
+    public function clear()
+    {
+        $this->points = [];
+        return $this;
+    }
+
+    /**
+     * Gets a filtered TimeLine, containing the TimePoint corresponding to the provided criteria
+     * @param string|array $tag A tag or a list of tags to filter
+     * @param int $target The type of target TimePoint to filter
+     * @param int $type The type of TimePoint to filter
+     * @return TimeLine Returns a subset corresponding to the found TimePoints
+     */
+    public function filter($tag = null, $target = TimePoint::TARGET_ALL, $type = TimePoint::TYPE_ALL)
+    {
+        // the tag criteria can be omitted
+        $tags = null;
+        if (isset($tag)) {
+            $tags = is_array($tag) ? $tag : [$tag];
+        }
+        
+        // create a another instance of the same class
+        $class = get_class($this);
+        $subset = new $class();
```

Comment:
not sure, but If I remember there's something like `new static()` introduced in php ~5.2 to create a new instance of the current class safely. (please verfiy)